### ISSUE TITLE
File version set to 0.0.0

### ISSUE THREAD

**Issue by davideforgione** [castleproject/Core#214]
Using nuget package i have this:

Assembly 'Castle.Windsor' with identity 'Castle.Windsor, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc' uses 'Castle.Core, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc' 
which has a higher version than referenced assembly 'Castle.Core' with identity 'Castle.Core, Version=0.0.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc'	

Castle.core dll is build with version 0.0.0.0, how's this?

If i build from sources i have the same realease version 0.0.0. To get the correct version build i had to manually run Set-VersionInfo.ps1 and rebuild package. Can someone upgrade nuget library please?

Thanks.

**Comment by davideforgione** [comment]
Added [assembly: SecurityTransparent()] to CommonAssemblyInfo.cs too to make it work with Castle.Windsor

**Comment by jonorossi** [comment]
I accidentally released 4.0.0-beta001 with version number of the .NET 4.5 assembly not set, it has been fixed in #206 but another package hasn't been released.

**Comment by nil4** [comment]
Are there any plans to release a new version that fixes this issue? It's been a few months now without an update.

**Comment by jonorossi** [comment]
I didn't think this was blocking anyone because I suspect Windsor 3.3.x probably isn't compatible with this version anyway, so didn't bother, didn't even know @AndriySvyryd thumbs up the issue either until now.

I'll get another package out in the next day.

**Comment by nil4** [comment]
@jonorossi the issue I'm seeing is due to [Moq 4.6.38-alpha](https://www.nuget.org/packages/Moq/4.6.38-alpha) that has dependencies on v3.3.3 for .NETFramework4.5 but v4.0.0-beta001 on .NETStandard1.3. As a result, when running tests for .NETFramework, I get this error:

```
The type initializer for 'Moq.Mock`1' threw an exception.
The type initializer for 'Moq.Proxy.CastleProxyFactory' threw an exception.
Could not load file or assembly 'Castle.Core, Version=0.0.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
Could not load file or assembly 'Castle.Core, Version=3.3.0.0, Culture=neutral, PublicKeyToken=407dd0808d44fbdc' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
   at Moq.Mock`1.<InitializeInstance>b__24_0()
   at Moq.PexProtector.Invoke(Action action)
   at Moq.Mock`1.InitializeInstance()
   at Moq.Mock`1.OnGetObject()
   at Moq.Mock.GetObject()
   at Moq.Mock.get_Object()
   at Moq.Mock`1.get_Object()
```

In contrast, [Moq 4.6.36-alpha](https://www.nuget.org/packages/Moq/4.6.36-alpha) depends on v4.0.0-beta001 in both frameworks and works as expected.

**Comment by jonorossi** [comment]
> Moq 4.6.38-alpha that has dependencies on v3.3.3 for .NETFramework4.5 but v4.0.0-beta001 on .NETStandard1.3

Oh, I had no idea the Moq guys changed to have different version ranges between targets. Sorry for the delay, trying to get #219 into the next release too.

**Comment by jonorossi** [comment]
Thanks for your patience, I've just released 4.0.0-beta002 with this change.
http://www.castleproject.org/blog/2016/10/28/core-4-0-0-beta002-release/

**Comment by nil4** [comment]
@jonorossi much appreciated! Explicitly adding this version to my test project.json forced Moq to use it and resolved the issue I was seeing.
